### PR TITLE
Fixed TS type errors

### PR DESCRIPTION
**Pull request by ajmeese7**
Several spots were having TS errors thrown by eslint unnecessarily, this commit works to address those errors.

Discord name: ajmeese7#4835

# Community PR Review Checklist

## First Time Specifics

- [ ] if its your first pull request to Dendron, watch out for the [CLA](https://en.wikipedia.org/wiki/Contributor_License_Agreement) bot that will ask you to agree to Dendron's CLA
- [ ] if its your first pull request and you're on our Discord, make sure that Kevin gives you the [horticulturalist](https://wiki.dendron.so/notes/7c00d606-7b75-4d28-b563-d75f33f8e0d7.html#horticulturalist) role by adding your discord as part of the description  👨‍🌾👩‍🌾
- [ ] (optional) ping `@Dendron Team` in the `#dev` channel of our discord - we usually respond to PRs within 24h

## Commit

- [x] make sure your branch names adhere to our [branch style](https://docs.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html#branch-style)
- [x] make sure the commit message follows Dendron's [commit style](https://docs.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html#commit-style)
- [x] if this pull request is addressing an existing issue, make sure to link this PR to the issue that it is resolving.
  - NOTE: make sure to read [Addressing open issues](https://docs.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html#addressing-open-issues)

## Code

- [x] code should follow [Code Conventions](https://docs.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html#branch-style)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](https://docs.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html#commit-style).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

## Tests

- [ ] [Write Tests](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#writing-tests) 
- [ ] [Confirm existing tests pass](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#executing-tests)
- [ ] [Confirm manual testing](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#manual-testing) 
- [ ] Common cases tested
- [ ] 1-2 Edge cases tested
- [ ] If your tests changes an existing snapshot, snapshots have been [updated](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#updating-test-snapshots)

## Docs
- [ ] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [ ] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](https://docs.dendron.so/notes/3489b652-cd0e-4ac8-a734-08094dc043eb.html) or [Packages](https://docs.dendron.so/notes/32cdd4aa-d9f6-4582-8d0c-07f64a00299b.html)

### CODE REVIEW

**Comment by jonathanyeung** [comment]
Hi @ajmeese7 - thanks a lot for the contribution. 

I'm not seeing these show up as eslint errors, and our build should fail if they were. 

These instances seem related to lodash, for example `!_.isUndefined(opts.note.config.global.enableChildLinks)` provides a TS type guard  to ensure `enableChildLinks` isn't undefined. 

If you run `yarn setup` at the root of your repo, does it make the errors go away?

File: packages/common-all/src/utils/index.ts [hunk 1]
```diff
@@ -872,7 +872,7 @@ export class ConfigUtils {
       opts.note.config.global &&
       !_.isUndefined(opts.note.config.global.enableChildLinks)
     ) {
-      return opts.note.config.global.enableChildLinks;
+      return opts.note.config.global.enableChildLinks!;
```
Comment (kevinslin) [inline packages/common-all/src/utils/index.ts:875]: what version of typescript are you using? this should be auto-inferred by typescript with the last line: `!_.isUndefined(opts.note.config.global.enableChildLinks)`

Comment (ajmeese7) [reply packages/common-all/src/utils/index.ts:875]: I installed the `package.json` as expected, so `4.5.5`

**Comment by ajmeese7** [comment]
> Hi @ajmeese7 - thanks a lot for the contribution.
> 
> I'm not seeing these show up as eslint errors, and our build should fail if they were.
> 
> These instances seem related to lodash, for example `!_.isUndefined(opts.note.config.global.enableChildLinks)` provides a TS type guard  to ensure `enableChildLinks` isn't undefined.
> 
> If you run `yarn setup` at the root of your repo, does it make the errors go away?

@jonathanyeung yep, running `yarn setup` fixed it! Thought I did that last time but I guess I was mistaken. Closing this out now, sorry about that :)

**Pull request closed without merging**